internal/controller/authenticator.go

```diff
@@ -181,11 +192,23 @@ func SetUpOAuthConnection(
 	}
 	tokenSource = GetTokenSource(ctx, oauthConfig, tlsConfig)
 	creds := credentials.NewTLS(tlsConfig)
+
+	proxyDialer := func(ctx context.Context, addr string) (net.Conn, error) {
+		proxyURL, err := http.ProxyFromEnvironment(&http.Request{URL: &url.URL{Host: addr}})
```

Comment:
You must define the HTTPS_PROXY environment variable in the Deployment's pod spec's `env` here:
https://github.com/illumio/cloud-operator/blob/df44b593931b44dd34b00bfa7b25eb248c9f90a5/cloud-operator/templates/cloud-operator-deployment.yaml#L56
Set it from a new Helm value, and define that value with `""` as default in `cloud-operator/values.yaml`.

Reply:
And once that's done, you should extend the `k8s_cluster` Terraform module to add a new `https_proxy` input variable here:
https://github.com/illumio/terraform-illumio-cloudsecure/blob/fc0e11de97687e8d74c4f0eb576027f5e898e0a1/modules/k8s_cluster/variables.tf
And use that variable to set the Helm value in the `helm_release` resource here:
https://github.com/illumio/terraform-illumio-cloudsecure/blob/fc0e11de97687e8d74c4f0eb576027f5e898e0a1/modules/k8s_cluster/main.tf#L5
Follow what we did for the `enable_falco` variable.

Reply:
Closing. The Terraform module work will be done in that repo.